docs/python/guides/flags.md

```diff
@@ -5,4 +5,238 @@ sidenav: side-nav-python.html
 type: markdown
 ---
 
-## Flags
\ No newline at end of file
+# Flags
+
+`absl.flags` defines a *distributed* command line system, replacing systems like
+`getopt()`, `optparse` and manual argument processing. Rather than an
+application having to define all flags in or near `main()`, each Python module
+defines flags that are useful to it. When one Python module imports another,
+it gains access to the other's flags. (This is implemented by having all modules
+share a common, global registry object containing all the flag information.)
+
+It includes the ability to define flag types (`boolean`, `float`, `integer`,
+`list`), autogeneration of help (in both human and machine readable format) and
+reading arguments from a file. It also includes the ability to automatically
+generate man pages from the help flags.
+
+Flags are defined through the use of one of the `DEFINE_xxx` functions. The
+specific function used determines how the flag is parsed, checked, and
+optionally type-converted, when it's seen on the command line.
+
+## Example Usage
+
+```python
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+
+# Flag names are globally defined!  So in general, we need to be
+# careful to pick names that are unlikely to be used by other libraries.
+# If there is a conflict, we'll get an error at import time.
+flags.DEFINE_string('name', 'Jane Random', 'Your name.')
+flags.DEFINE_integer('age', None, 'Your age in years.', lower_bound=0)
+flags.DEFINE_boolean('debug', False, 'Produces debugging output.')
+flags.DEFINE_enum('job', 'running', ['running', 'stopped'], 'Job status.')
+
+
+def main(argv):
+  if FLAGS.debug:
+    print('non-flag arguments:', argv)
+  print('Happy Birthday', FLAGS.name)
+  if FLAGS.age is not None:
+    print('You are %d years old, and your job is %s' % (FLAGS.age, FLAGS.job))
+
+
+if __name__ == '__main__':
+  app.run(main)
+```
+
+## Flag Types
+
+This is a list of the `DEFINE_*`'s that you can do. All flags take a name,
+default value, help-string, and optional 'short' name (one-letter name). Some
+flags have other arguments, which are described with the flag.
+
+*   `DEFINE_string`: takes any input, and interprets it as a string.
+*   `DEFINE_bool` or `DEFINE_boolean`: typically does not take an argument: say
+    `--myflag` to set `FLAGS.myflag` to `True`, or `--nomyflag` to set
+    `FLAGS.myflag` to `False`. `--myflag=true` and `--myflag=false` are also
+    supported, but not recommended.
+*   `DEFINE_float`: takes an input and interprets it as a floating point number.
+    Takes optional args lower_bound and upper_bound; if the number specified on
```

Comment:
args lower_bound and upper_bound -> arguments `lower_bound` and `upper_bound`

I'd avoid terms like "args" unless it refers to actual code.